src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs

```diff
@@ -22,7 +22,7 @@ internal class AuthorityInfo
             AuthorityType authorityType,
             string authority,
             bool validateAuthority)
-            : this(authorityType, new Uri(authority), validateAuthority)
+            : this(authorityType, ValidateAndCreateAuthorityUri(authority, authorityType), validateAuthority)
```

Comment:
This is the main change.

Reply:
Adding some comments in the `ValidateAndCreateAuthorityUri` method for each of the checks we are doing will be good to know the validation coverage we have.